DataStructure1/assignment_1/ArrayList.java

```diff
@@ -0,0 +1,263 @@
+
+package assignment_1;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.NoSuchElementException;
+
+public class ArrayList<E> 
+{
+	private int size;
+	// an object variable for array
+	private Object[] array;
+	public ArrayList(int initialCapacity)
+	{
+		if(initialCapacity < 0)
+			throw new IllegalArgumentException("Illegal Capacity: "+initialCapacity);
+		this.array = new Object[initialCapacity];
+	}
+
+	
+	public ArrayList()
+	{
+		this.array = new Object[10];
+	}
+
+	public E add(E e)
+	{
+		// to ensure the size of mergedList
+		ensureCapacity(size+1);
```

Comment:
This statement creates new Array if the array size is not full. Please correct the statement.